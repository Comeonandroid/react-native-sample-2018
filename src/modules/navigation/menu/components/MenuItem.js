import React, { PureComponent } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { ifProp } from 'styled-tools'

import CustomIcon from '../../../../utils/icon'

import { Text, TouchableOpacity } from '../../../../elements'

class MenuItem extends PureComponent {

  constructor(params) {
    super(params)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    const { route, onPress } = this.props
    if(onPress) { onPress(route) }
  }

  render() {
    return (
      <Touchable noBorder={this.props.noBorder} onPress={this.handleClick}>
        <Label>{this.props.children}</Label>
        <MenuIcon name={this.props.icon} size={this.props.size}></MenuIcon>
      </Touchable>
    )
  }
}

MenuItem.propTypes = {
  children: PropTypes.node.isRequired,
  icon: PropTypes.string.isRequired,
  route: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  size: PropTypes.number,
  noBorder: PropTypes.bool
}

MenuItem.defaultProps = {
  size: 16
}

// TODO: last child border
const Touchable = styled(TouchableOpacity) `
  padding-bottom: 14px;
  padding-top: 14px;
  borderBottomWidth: ${ifProp({ noBorder: true }, 0, '1px')}
  borderBottomColor: ${ ({ theme })  => theme.colors.drawerItemBorderColor };
  display: flex;
  flex-direction: row;
  align-items: center;
`

const Label = styled(Text) `
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0.6px;
  color: ${({ theme }) => theme.colors.drawerContrastColor }
`

const MenuIcon = styled(CustomIcon) `
  color: ${({ theme }) => theme.colors.drawerIconColor};
  margin-left: auto;
`

export default MenuItem
