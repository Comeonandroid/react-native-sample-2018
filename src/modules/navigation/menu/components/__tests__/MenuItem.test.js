import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../../../specs/shared_examples/index'

import MenuItem from '../MenuItem'

const setup = propOverrides => {
  const props = Object.assign({
    children: 'trade',
    icon: 'trade',
    route: 'Trade',
    onPress: jest.fn()
  }, propOverrides)

  const component = <MenuItem {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<MenuItem />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  it('onPress: properly fires when button pressed', () => {
    const { props, wrapper } = setup()
    wrapper.simulate('press')
    expect(props.onPress).toHaveBeenCalledTimes(1)
    expect(props.onPress).toHaveBeenCalledWith(props.route)
  })
})
