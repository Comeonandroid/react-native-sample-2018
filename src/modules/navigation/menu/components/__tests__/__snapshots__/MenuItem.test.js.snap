// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> dark: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#3c3c3c",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#c1c1c1",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.6,
          },
          undefined,
        ],
      ]
    }
  >
    trade
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "color": "#c1c1c1",
              "marginLeft": "auto",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`<MenuItem /> light: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#716d6d",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "color": "#edecec",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.6,
          },
          undefined,
        ],
      ]
    }
  >
    trade
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "color": "rgb(255,255,255)",
              "marginLeft": "auto",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
